### PR TITLE
Gatsby tooling

### PR DESCRIPTION
**Pull request by AndrewJDick**
Added basic tooling to the site:
- Lint-Staged and Husky for pre-commit linting. 
- ESLint (extending react-app) and `.eslintrc`.
- Prettier, `.prettierrc`, and `prettierignore`.
- Updated `.editorconfig`

Closes: https://github.com/fathomlondon/fathomlondon.com/issues/187

### CODE REVIEW

**Comment by peduarte** [comment]
Deploy preview for *fathomlondon* ready!

Built with commit b7fc97312bba6a87d9af135b77c3bda2f4a0c08b

https://deploy-preview-194--fathomlondon.netlify.com

File: .editorconfig [hunk 1]
```diff
@@ -3,11 +3,12 @@ root = true
 # Apply for all files
 [*]
 charset = utf-8
-
+indent_size = 2
```
Comment (artisologic) [inline .editorconfig]: I feel this one doesn't make sense if you use `indent_style = tab`. I usually omit it.

File: package.json [hunk 2]
```diff
@@ -1,26 +1,46 @@
...
+  "version": "1.0.0",
+  "description": "Fathom's website",
+  "author": "Fathom London",
+  "main": "n/a",
+  "scripts": {
+    "start": "npm run develop",
+    "build": "gatsby build",
+    "develop": "gatsby develop",
+    "precommit": "lint-staged",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (artisologic) [inline package.json]: Do we need to add a convenience script here to reformat the whole codebase in case prettier updates?

File: package.json [hunk 3]
```diff
@@ -1,26 +1,46 @@
...
+    "gatsby": "^1.9.183",
+    "gatsby-plugin-react-helmet": "^2.0.4",
+    "gatsby-plugin-sass": "^1.0.16",
+    "react-helmet": "^5.2.0"
+  },
+  "devDependencies": {
+    "babel-eslint": "^7.2.3",
+    "eslint": "^4.1.1",
+    "eslint-config-react-app": "^2.1.0",
+    "eslint-plugin-flowtype": "^2.34.1",
```
Comment (artisologic) [inline package.json]: Do we need this as we don't use flow? Or was it just because `eslint-config-react-app` moans because it's a peerDep?

File: src/_sass/_flexbox.scss [hunk 4]
```diff
@@ -1,55 +1,135 @@
...
-.order-6#{$suffix} { order: 6; }
-.order-7#{$suffix} { order: 7; }
-.order-8#{$suffix} { order: 8; }
-.order-last#{$suffix} { order: 99999; }
+	flex: 1 1 auto;
+	min-width: 0; /* 1 */
+	min-height: 0; /* 1 */
+}
+
+.flex-none#{$suffix} {
```
Comment (artisologic) [inline src/_sass/_flexbox.scss]: might want to check if @peduarte is happy with the css formatting here. Not sure if that was the intention or not. Personally I'm ok with letting prettier do its thing here, even though arguably it might be less readable for a css codebase in that style.

File: package.json (hunk 2 above)
Comment (peduarte) [reply package.json]: Yes, I usually call this a `format` script.

File: package.json (hunk 3 above)
Comment (peduarte) [reply package.json]: Yeah, it's required by [eslint-config-fbjs](https://www.npmjs.com/package/eslint-config-fbjs)

On that note, didn't we say we'd use the [eslint-config-react-app](https://www.npmjs.com/package/eslint-config-react-app)?

File: src/_sass/_flexbox.scss (hunk 4 above)
Comment (peduarte) [reply src/_sass/_flexbox.scss]: @artisologic what do you mean exactly?

File: package.json (hunk 3 above)
Comment (artisologic) [reply package.json]: that's what @AndrewJDick used.

File: src/_sass/_flexbox.scss (hunk 4 above)
Comment (artisologic) [reply src/_sass/_flexbox.scss]: I mean that your neatly one-liner css was reformatted by prettier as multi-line.
Just wanted to check if you were ok with that.

Comment (peduarte) [reply src/_sass/_flexbox.scss]: Oh yeah, I'm cool with that.

File: package.json (hunk 3 above)
Comment (peduarte) [reply package.json]: Oh yeah 🤦‍♂️ 

`eslint-config-react-app` is required by `eslint-config-react-app` too.

Comment (artisologic) [reply package.json]: image

lol

Comment (peduarte) [reply package.json]: Oh man, I'm going to bed.

Comment (AndrewJDick) [reply package.json]: All of them are Peer dependencies, found here:
https://www.npmjs.com/package/eslint-config-react-app

File: src/_sass/_flexbox.scss (hunk 4 above)
Comment (AndrewJDick) [reply src/_sass/_flexbox.scss]: Yeah, I briefly looked into this. StyleLint had a single line declaration rule, but it's been deprecated. I can't see anything similar on Prettier, other than perhaps extending the print width per line? Feels like more hassle than what it's worth.

File: package.json (hunk 2 above)
Comment (AndrewJDick) [reply package.json]: Added `"format": "prettier --config .prettierrc --write \"src/**/*\""`

File: .eslintrc [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+  'extends': 'react-app',
```
Comment (peduarte) [inline .eslintrc]: Should we also extend [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier)?

Comment (peduarte) [reply .eslintrc]: and [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier)? 

@artisologic what do you think?

Comment (AndrewJDick) [reply .eslintrc]: We can actually import both in a single extend
https://prettier.io/docs/en/eslint.html

File: .eslintrc [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-    'plugin:prettier/recommended'
   ],
 
   'rules': {
     // graphQL
     'relay/graphql-naming': 0,
 
     // react
-    'react/forbid-prop-types': ['error', { forbid: ['any'] }],
+    'react/prop-types': 1,
```
Comment (peduarte) [inline .eslintrc]: Are we not going to use `eslint-config-prettier` in conjunction with `eslint-config-react-app`?

File: src/layouts/index.js [hunk 7]
```diff
@@ -15,12 +15,12 @@ function Layout({ data, children }) {
...
-`;
+// export const pageQuery = graphql`
+// 	query layoutQuery {
+// 		site {
+// 			siteMetadata {
+// 				title
+// 			}
+// 		}
+// 	}
+// `;
```
Comment (peduarte) [inline src/layouts/index.js]: Why is this commented out?

File: src/pages/index.js [hunk 8]
```diff
@@ -1,7 +1,12 @@
...
-	return <h1>{foo} Hello World.</h1>;
+function Index({ foo }) {
+	return <h2>{foo} Hello World.</h2>;
 }
 
 export default Index;
+
+// Index.propTypes = {
+// 	foo: PropTypes.any,
+// };
```
Comment (peduarte) [inline src/pages/index.js]: Can we delete commented out code?

**Comment by peduarte** [comment]
@AndrewJDick Left a few comments. @artisologic this is ready for a review. @jjenzz would you mind have a look too?

File: src/layouts/index.js (hunk 7 above)
Comment (AndrewJDick) [reply src/layouts/index.js]: I was testing eslint on prepush. Apologies

File: src/layouts/index.js [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
+import React from 'react';
+import Helmet from 'react-helmet';
+import './index.scss';
 
 function Layout({ data, children }) {
 	return (
 		<div>
 			<Helmet>
-				<title>{data.site.siteMetadata.title}</title>
+				<span>{data.site.siteMetadata.title}</span>
```
Comment (jfidiles) [inline src/layouts/index.js]: Why is this changed from `title` to `span`?

File: .prettierrc [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{
+	"useTabs": true,
+	"singleQuote": true,
+	"trailingComma": "es5",
+	"printWidth": 80
```
Comment (peduarte) [inline .prettierrc]: We don't need this as it's default

**Comment by peduarte** [comment]
@artisologic @jjenzz @lukemartin @jfidiles Ready for review

File: src/layouts/index.js [hunk 11]
```diff
@@ -28,6 +29,12 @@ function Layout({ data, children }) {
 
 export default Layout;
 
+Layout.propTypes = {
```
Comment (jfidiles) [inline src/layouts/index.js]: nice

Comment (AndrewJDick) [reply src/layouts/index.js]: image
